src/main/java/org/json/JSONObject.java

```diff
@@ -2892,4 +2922,22 @@ private static JSONException recursivelyDefinedObjectException(String key) {
             "JavaBean object contains recursively defined member variable of key " + quote(key)
         );
     }
+
+    /**
+     * For a prospective number, remove the leading zeros
+     * @param value prospective number
+     * @return number without leading zeros
+     */
+    private static String removeLeadingZerosOfNumber(String value){
+        if (value.equals("-")){return value;}
+        boolean negativeFirstChar = (value.charAt(0) == '-');
+        int counter = negativeFirstChar ? 1:0;
+        while (counter < value.length()){
+            if (value.charAt(counter) != '0'){
+                return String.format("%s%s", negativeFirstChar?'-':"",value.substring(counter));
```

Comment:
I'm still not a fan of `String.format` here.

```java
if (value.charAt(counter) != '0') {
   if (negativeFirstChar) return "-"+value.substring(counter);
   return value.substring(counter);
}
++counter;
```

Reads better and doesn't require the extra overhead of `String.format`

Reply:
If you REALLY hate using the `+` operator on strings, my last recommendation would be the following:

```java
while (counter < value.length()){
    if (value.charAt(counter) != '0') {
        if (negativeFirstChar) return "-".concat(value.substring(counter));
        return value.substring(counter);
    }
    ++counter;
}
if (negativeFirstChar) return "-".concat(value.substring(counter));
return value.substring(counter);
```

Reply:
Fair enough. I really feel String concatenation using overloaded "+" operator, makes the code much more unreadable.
I do agree, that there is an overhead of String formatting, but it gives the coder a clear picture of the end result.
Concat wins the middle ground, and I have implemented it.